test/check_event_log.txt

```diff
@@ -0,0 +1,19 @@
+; Checks the pattern of event log file. Makes sure the data rows
+; Corresponding to each layer have the correct format.
+; Checks the shader hashes across different rows and guarantees they are
+; consistent.
+; Counts the number of memory and frame time logs and check if they are
+; as expected.
+; CHECK-DAG:  compile_time_layer_init,{{[0-9]+}}
+; CHECK-DAG:  memory_usage_layer_init,{{[0-9]+}}
+; CHECK-DAG:  frame_time_layer_init,{{[0-9]+}}
+; CHECK:      create_shader_module_ns,{{[0-9]+}},[[SHADER1:0x[a-zA-Z0-9]+]],{{[0-9]+}}
+; CHECK-NEXT: create_shader_module_ns,{{[0-9]+}},[[SHADER2:0x[a-zA-Z0-9]+]],{{[0-9]+}}
+; CHECK:      shader_module_first_use_slack_ns,{{[0-9]+}},[[SHADER1]],{{[0-9]+}}
+; CHECK-NEXT: shader_module_first_use_slack_ns,{{[0-9]+}},[[SHADER2]],{{[0-9]+}}
+; CHECK:      create_graphics_pipelines,{{[0-9]+}},"[[[SHADER1]],[[SHADER2]]]",{{[0-9]+}}
+; CHECK-DAG:  memory_usage_present,{{[0-9]+}},{{[0-9]+}},{{[0-9]+}}
+; CHECK-DAG:  frame_present,{{[0-9]+}},{{[0-9]+}},{{[0-9]+}}
+; CHECK-NOT:  NOT
```

Comment:
Why do we need this?

Reply:
To make sure memory_destroy_device and frame_time_layer_exit happen after memory_usage_present and frame_present.

Reply:
Now that you mentioned it, It seems a bit untidy to me as well. I changed it.